### DISCUSSION TITLE
testing and global entity manager

### DISCUSSION THREAD

**Discussion by xenoterracide** [mikro-orm/nestjs#90] [Q&A]
```
ValidationError: Using global EntityManager instance methods for context specific actions is disallowed. If you need to work with the global instance's identity map, use `allowGlobalContext` configuration option or `fork()` instead.
```

allow Global Context seems bad, but I can't figure out how I should be using a different context in a jest test which only has repositories and entities.

**Comment by B4nan** [comment]
It's fine for tests. Alternative is to fork or use `RequestContext` helper explicitly.

**Reply by xenoterracide** [reply]
ok, but there's no real documented path for doing this, and I can't figure out how to for when injecting a (not custom) repository into a service.

**Reply by B4nan** [reply]
https://mikro-orm.io/docs/usage-with-nestjs#testing-1

**Reply by xenoterracide** [reply]
I'm not mocking, I'm using the real thing against an in memory mocking

**Reply by B4nan** [reply]
I have no idea what you mean by that.

**Reply by xenoterracide** [reply]
there is no mock, nothing is getting replaced.

urgh, I meant to say "in memory database"
